javaparser-core/src/main/java/com/github/javaparser/ast/StubUnit.java

```diff
@@ -0,0 +1,153 @@
+        super(null);
+        this.compilationUnits = compilationUnits;
+    }
+
+    /** The getter method for the list of the compilation units of the stub file. */
+    public List<CompilationUnit> getCompilationUnits() {
+        return compilationUnits;
+    }
+
+    /** The setter method for the list of the compilation units of the stub file. */
+    public void setCompilationUnits(NodeList<CompilationUnit> compilationUnits) {
+        this.compilationUnits = compilationUnits;
+    }
+
+    /** @return information about where this stub unit was loaded from, or empty if it wasn't loaded from a file.*/
+    public Optional<StubUnit.Storage> getStorage() {
+        return Optional.ofNullable(storage);
+    }
+
+    public StubUnit setStorage(Path path) {
+        this.storage = new Storage(this, path);
+        return this;
+    }
+
+    //TODO add a visit method
+    @Override
+    public <R, A> R accept(GenericVisitor<R, A> v, A arg) {
+        throw new RuntimeException("The method is not implemented!");
+         //v.visit(this, arg);
+    }
+
+    //TODO add a visit method
+    @Override
+    public <A> void accept(VoidVisitor<A> v, A arg) {
+        throw new RuntimeException("The method is not implemented!");
+        // v.visit(this, arg);
+    }
+
+    /**
+     * Information about where this stub unit was loaded from.
+     * This class only stores the absolute location.
+     * For more flexibility use SourceRoot.
+     */
+    public static class Storage {
+
+        /** An stub unit that it represents. */
+        private final StubUnit stubUnit;
+
+        /** The path to the source for this stub unit. */
+        private final Path path;
+
+        /**
+         * The constructor with all fields.
+         *
+         * @param stubUnit is the stub unit that it describes.
+         * @param path to the source for this stub unit.
+         */
+        private Storage(StubUnit stubUnit, Path path) {
+            this.stubUnit = stubUnit;
+            this.path = path.toAbsolutePath();
+        }
+
+        /** @return the path to the source for this StubUnit. */
+        public Path getPath() {
+            return path;
+        }
+
+        /** @return the StubUnit this Storage is about. */
+        public StubUnit getStubUnit() {
+            return stubUnit;
+        }
+
+        /** @return the file name of the stub file that represented by the StubUnit. */
+        public String getFileName() {
+            return path.getFileName().toString();
+        }
+
+        /** @return the directory with the stub file that represented by the StubUnit. */
+        public Path getDirectory() {
+            return path.getParent();
+        }
+
+        /** Saves the stub unit to its original location.*/
+        public void save() {
+            save(cu -> new PrettyPrinter().print(getStubUnit()));
+        }
+
+        /** Saves the stub unit to its original location and give the output.*/
```

Comment:
What does "give the output" mean?
How is `makeOutput` used?
(I realize that these things are not documented in CompilationUnit.java either.  It would be nice to create a pull request against upstream that adds documentation to clarify it there, since you have figured it out.)